### ISSUE TITLE
CSS for Sharing not applied if from template

### ISSUE THREAD

**Issue by serrapa** [TriliumNext/Trilium#3562]
### Trilium Version

0.58.7

### What operating system are you using?

macOS

### What is your setup?

Local + server sync

### Operating System Version

MacOS Monterey 12.4

### Description

I have a note with the #template label and the ~shareCss(inheritable)=<file_css>. I noticed that every note created with that template relation do not get the css from the file_css once being shared. It seems the ~shareCss is taken in consideration only when is on the owned attributes and not on inherited ones.

**Comment by sigaloid** [comment]
This is intended behavior AFAIK, you can click the attribute to check off inheritable from within the template.

**Comment by serrapa** [comment]
Yeah I know I can check it off, but I would need that all notes get the css style from the specified file in the ~shareCss. Basically, the app should consider also the inherited attributes for the shareCss when creating the sharing note. I dont know if this is the intended behaviour for some reason, but if there are no problems in doing it, why not?

**Comment by zadam** [comment]
Hi, `shareCss` should be inheritable as any other attribute. I can't reproduce the issue unfortunately.

One question - is the template part of the shared subtree?

**Comment by serrapa** [comment]
Hey @zadam, the shared note has the ~template relation. To reproduce the issue try to:
- create a <css_note> file with the following content and share it:
`#content h2 {color: red;}`
- create a template note A with the following attributes: **_#template=A , ~shareCss(inheritable)=<css_note>_**)
- create another note B with the relation to the template (_**~template=A**_) and a content with at least one Heading 2. It gets the **_~shareCss=<css_note>_**  as inherited attribute (from A).
- create other notes inside B (this is my case, but I suppose it's not relevant this step)
- share the note B and open the link
- see that h2 tag are not red.

Now to make things work:
- try to add directly the relation **_~shareCss(inheritable)=<css_note>_** in note B as owned attribute and see that the h2 are now red.

I don't know if I am doing somethings wrong or not. If so, please tell what it is!

**Comment by zadam** [comment]
So your tree looks something like this:

* A (template)
* B (~template=A)
    *  C
    *  D

And then you share B. The problem is that when you share B, you only publish B and its subtree. A is outside of this subtree and thus not published, which makes it inaccessible to the shared website - the content, attributes etc. are not used in any way.

This might seem strict, but the point is to make it very clear what is shared and what not to avoid situations where you might unintentionally link to some information you don't want public.

**Comment by serrapa** [comment]
Ahh now I understood, I miss the part where I needed to share also the A template! I totally forgot it! 

Anyway, it is not very easy to handle sharing in case of templates. If you share your template A, the note B is also shared (the shared button is on and you cannot change it by switching from on to off). Since this, if you click on the link of B, it returns 404 Not found. That's because you need to share B before and then share A. Try it and lemme know

**Comment by zadam** [comment]
Huh, so this was a bug - some components wrongly reported that B is shared which made it impossible to toggle sharing, but still returned 404 (because it wasn't really shared).

Thanks for reporting this, I fixed it and it will be in the next patch release.
